examples/translation_example.py

```diff
@@ -0,0 +1,483 @@
+        ddf["word_count"] = ddf["word_count"].astype("int64")
+        ddf_true = ddf[(ddf["word_count"] <= self.translation_config.max_words_per_sen)]
+
+        ddf_metaa = ddf_true._meta.copy()
+        ddf_metaa["isalpha"] = False
+        # To filter for atleast one unicode letter in text
+        has_letter = ddf_true.map_partitions(self.atleast_letter, meta=ddf_metaa)
+        ddf_trans = ddf_true[has_letter["isalpha"]]
+        ddf = ddf_trans.drop(columns="word_count")
+
+        ## ddf false operations
+        ddf_false = ddf_true[~has_letter["isalpha"]]
+        ddf_false = ddf_false.drop(columns="word_count")
+        ddf_false_meta = ddf_false._meta.copy()
+        ddf_false_meta["translation"] = ""
+        ddf_false = ddf_false.map_partitions(
+            self.combine_text_false, meta=ddf_false_meta
+        )
+
+        # Applying preprocess_df for Indic preprocessing
+        ddf["indic_proc_text"] = ddf["indic_proc_text"].astype("str")
+        ddf = ddf.map_partitions(self.preprocess_df, meta=ddf_meta)
+
+        columns = ddf.columns.tolist()
+        pipe = op.Sequential(
+            op.Tokenizer(
+                model, cols=[self.input_column], tokenizer_type="sentencepiece"
+            ),
+            op.Predictor(
+                self.model,
+                sorted_data_loader=True,
+                batch_size=self.batch_size,
+                pred_output_col="translation",
+            ),
+            keep_cols=columns,
+        )
+        ddf = pipe(ddf)
+        translated_meta = ddf._meta.copy()
+        translated_meta["translation"] = "DUMMY_STRING"
+        ddf = ddf.map_partitions(self.translate_tokens, meta=translated_meta)
+        ddf = ddf.map_partitions(self.combine_text, meta=translated_meta)
+
+        # Merging translated and non-translated samples
+        ddf_true["false_translation"] = ddf_false["translation"]
+        ddf_true["false_translation"] = ddf_true["false_translation"].fillna("")
+        ddf_true["translation"] = ddf["translation"]
+        ddf_true["translation"] = ddf_true["translation"].fillna("")
+        ddf_true["translation"] = (
+            ddf_true["translation"] + ddf_true["false_translation"]
+        )
+
+        ddf = ddf_true.map_partitions(self.grouping)
+        write_to_disk(
+            ddf,
+            output_file_dir=self.output_data_dir,
+            write_to_filename=True,
+            output_type=self.output_file_type,
+        )
+
+        print(f"Total time taken for translation: {time.time()-st} seconds", flush=True)
+
+
+def main(args):
+    print(f"Arguments parsed = {args}")
+    client = get_client(**parse_client_args(args))
+    print(client.dashboard_link)
+    it = IndicTranslation(
+        args.input_data_dir,
+        args.output_data_dir,
+        args.pretrained_model_name_or_path,
+        args.input_file_type,
+        args.output_file_type,
+        args.input_column,
+        args.batch_size,
+        args.autocast,
+    )
```

Comment:
You should pass in the DocumentDataset here , see below example again to follow instead of `args.input_data_dir` and `args.output_data_dir` here .  IndicTranslation should just take Model Related inputs and nothing related to input or output data.   

https://github.com/NVIDIA/NeMo-Curator/blob/main/examples/classifiers/aegis_example.py

This class should look something like below: 

```python3

translator_model = IndicTranslation(
        args.pretrained_model_name_or_path,
        args.batch_size,
        args.autocast
  )

input_dataset  = DocumentDataset.read_json(
      input_file_path, backend="cudf", add_filename=True
  )
result_dataset = translator_model(dataset=input_dataset)

result_dataset.to_json(output_file_dir=output_file_path, write_to_filename=True)

 ```

Reply:
Added this logic. Thanks.